src/documents/ContextAwareness.md

```diff
@@ -0,0 +1,327 @@
+fluid.defaults("fluid.uploader.compatibility.1_2", {
+    transformOptions: {
+        transformer: "fluid.model.transformWithRules",
+        config: fluid.compat.fluid_1_2.uploader.optionsRules
+    }
+});
+```
+
+The defaults block `fluid.uploader.compatibility.distributor.1_3` contains an options distribution which appeals to the existence of a third dimension in the uploader's `contextAwareness`, named
+`apiCompatibility` &#8212; this can be done simply by arranging to broadcast the appropriate options into it. Once we have defined this options distribution, we actually need to construct a component
+a component instance which holds and operates them &#8212; this is done via the `fluid.constructSingle` line. This utility automatically arranges for a singleton instance, uniquified at the 
+component tree's top level by the type `singleRootType` which has a very similar function to the option of the same name consumed by the `fluid.resolveRootSingle` grade described in
+the documentation on [Contexts](Contexts.md#global-components-fluid.resolveRoot-and-fluid.resolveRootSingle).
+
+Having shown the basic operation of the _receiver_ of contextual information, we'll now describe the group of utilities, including `fluid.constructSingle` that we just met, which can be
+used by integrators and implementors to coordinate the visibility of context names and distributions from them.
+
+Note that the combined effect of the first two defaults blocks shown in this example can be achieved "all-in-one" by a single call to the dedicated utility `fluid.contextAware.makeAdaptation`. 
+
+## Making contexts visible and removing them with `fluid.contextAware.makeChecks` and `fluid.contextAware.forgetChecks`
+
+The `checkRecord` structures described in the table above, by default appeal to the existence of some components holding the boolean value <code>true</code> at an option named <code>value</code> - or
+else some other value held at this path which is to be compared by equality. The `ContextAwareness` API includes two helper functions to assist integrators in issuing contexts of this form, and
+removing them when they are no longer required.
+
+These contexts can be issued with a call of the form:
+
+```
+fluid.contextAware.makeChecks(<checkStructure>);
+```
+
+The `checkStructure` argument is a hash of `contextName` strings to a `checkEntry` record which is documented in the following table:
+
+
+<table>
+    <thead>
+        <tr>
+            <th colspan="3">Members of a <code>checkEntry</code> record supplied as an argument to <code>fluid.contextAware.makeChecks</code></th>
+        </tr>
+        <tr>
+            <th>Member</th>
+            <th>Type</th>
+            <th>Description</th>
+        </tr>
+    </thead>
+    <tbody>
+        <tr>
+            <td><code>func</code>/<code>funcName</code></td>
+            <td><code>String</code> <a href="IoCReferences.md">IoC reference</a></td>
```

Comment:
The member name is given as "func/funcName" and the description says "a function or function name"; should the Type say "A function or a String IoC reference" ?